sdk/testing/src/test/java/io/opentelemetry/sdk/testing/assertj/TraceAssertionsTest.java

```diff
@@ -496,4 +499,59 @@ void optionalAttributes() {
                         satisfies(COINS, val -> val.containsExactly(0.01, 0.05, 0.1))))
         .isInstanceOf(AssertionError.class);
   }
+
+  private static TestSpanData buildTestSpan(String spanId, String spanName) {
+    return TestSpanData.builder()
+        .setResource(RESOURCE)
+        .setInstrumentationScopeInfo(INSTRUMENTATION_SCOPE_INFO)
+        .setName(spanName)
+        .setKind(SpanKind.CLIENT)
+        .setStartEpochNanos(100)
+        .setStatus(StatusData.ok())
+        .setEndEpochNanos(200)
+        .setHasEnded(true)
+        .setSpanContext(SpanContext.create(TRACE_ID, spanId, TraceFlags.getSampled(), TRACE_STATE))
+        .build();
+  }
+
+  @Test
+  void hasSpansSatisfyingExactly() {
+    Collection<List<SpanData>> traces = new ArrayList<>();
+    traces.add(Arrays.asList(buildTestSpan(SPAN_ID1, "span1"), buildTestSpan(SPAN_ID2, "span2")));
+
+    // test asserting spans in exact oder
+    TracesAssert.assertThat(traces)
+        .hasTracesSatisfyingExactly(
+            trace ->
+                trace.hasSpansSatisfyingExactly(
+                    span -> span.hasSpanId(SPAN_ID1), span -> span.hasSpanId(SPAN_ID2)));
+    // test asserting spans in wrong oder
+    assertThatThrownBy(
```

Comment:
Can we add `.isInstanceOf(AssertionError.class)` for the `assertThatThrownBy` lines?

Reply:
Done